sdk/samples/iot/sample_pnp_component_mqtt.c

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "sample_pnp_component_mqtt.h"
+
+#include <stdint.h>
+
+#include <azure/core/az_span.h>
+
+static uint32_t request_id_int;
+static char request_id_buf[8];
+
+// Create request id span which increments request id integer each call. Capable of holding 8 digit
+// number.
+az_span get_request_id(void)
+{
+  az_span remainder;
+  az_span out_span = az_span_init((uint8_t*)request_id_buf, sizeof(request_id_buf));
+
+  // Note that if left to run for a long time, this will overflow and reset back to 0.
```

Comment:
Do we want to increase `request_id_buf` to be of size 10 or 16 to cover all the bases (need 10, but a power of 2 might make sense)? `1_000_000_000` won't fit.

Or just add a comment saying this is for illustration purposes and we expect `request_id_int` to be a small number (for example, less than 1_000).

Reply:
makes sense increased to 10